src/types.ts

```diff
@@ -4,17 +4,18 @@ import { Context } from "./context";
 
 export type LiteralUnionStr<T extends U, U = string> = T | (string & object);
 
-export type FunctionTarget = {
-  [key: string]: Handler;
-};
+// It is not possible to get the generic type of a function in TypeScript.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type FunctionTarget = any;
 
 export type Controller = {
-  prototype: FunctionTarget;
   name: string;
   length: number;
   new (...args: unknown[]): unknown;
 };
 
+export type ClassConstructor<T = object> = new (...args: unknown[]) => T;
+
```

Comment:
I have changed definition of the controller like you have wanted but i didn't remove it because Constructor is different from Controller if i change it and if we want to add new property for controller in the future we have to make big changes whole of the package.